map/src/main/kotlin/edu/artic/map/MapViewModel.kt

```diff
@@ -0,0 +1,200 @@
+package edu.artic.map
+
+import com.fuzz.rx.asObservable
+import com.fuzz.rx.bindTo
+import com.fuzz.rx.disposedBy
+import edu.artic.db.daos.ArticGalleryDao
+import edu.artic.db.daos.ArticMapAnnotationDao
+import edu.artic.db.daos.ArticObjectDao
+import edu.artic.db.models.ArticGallery
+import edu.artic.db.models.ArticObject
+import edu.artic.viewmodel.BaseViewModel
+import io.reactivex.Observable
+import io.reactivex.functions.BiFunction
+import io.reactivex.functions.Function3
+import io.reactivex.functions.Function4
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.Subject
+import timber.log.Timber
+import javax.inject.Inject
+
+class MapViewModel @Inject constructor(
+        private val mapAnnotationDao: ArticMapAnnotationDao,
+        private val galleryDao: ArticGalleryDao,
+        private val objectDao: ArticObjectDao
+) : BaseViewModel() {
+
+    val mapAnnotations: Subject<List<MapItem<*>>> = BehaviorSubject.create()
+
+    val alwaysVisibleAnnotations: Subject<List<MapItem.Annotation>> = BehaviorSubject.create()
+
+    val floor: Subject<Int> = BehaviorSubject.createDefault(1)
+    val zoomLevel: Subject<MapZoomLevel> = BehaviorSubject.create()
+
+    val currentFloor: Int
+        get() {
+            val castFloor = (floor as BehaviorSubject)
+            return if (castFloor.hasValue()) {
+                castFloor.value
+            } else {
+                1
+            }
+        }
+
+    val currentZoomLevel: MapZoomLevel
+        get() {
+            val castLevel = (zoomLevel as BehaviorSubject)
+            return if (castLevel.hasValue()) {
+                castLevel.value
+            } else {
+                MapZoomLevel.One
+            }
+        }
+
+    init {
+        val allAmenities = mapAnnotationDao
+                .getAmenitiesOnMap()
+                .map { annotationList ->
+                    val list = mutableListOf<MapItem.Annotation>()
+                    annotationList.forEach { annotation ->
```

Comment:
can we just use ` map{}` call here? just map annotation to `MapItem.Annotation`. if you need to manually add to a list you can probably convert it to be more idiomatic kotlin.

Reply:
just do:
```
annotationList.map { annotation ->
  val floor = annotation.floor?.toInt() ?: 1
  return MapItem.Annotation(annotation, floor)
 }
```
but this seems like you are doing this conversion in many other places. make an extension on the `annotation` class to convert it.

Reply:
done